core_lib/src/structure/object.cpp

```diff
@@ -451,11 +451,26 @@ void Object::paintImage(QPainter& painter, int frameNumber,
             if (layer->type() == Layer::BITMAP)
             {
                 LayerBitmap* layerBitmap = (LayerBitmap*)layer;
-                layerBitmap->getLastBitmapImageAtFrame(frameNumber, 0)->paintImage(painter);
+
+                // Make sure there's a key before painting anything
+                // otherwise return
+                if (!layer->keyExists(frameNumber)) {
+                    return;
+                }
+
```

Comment:
I'm afraid this condition check is not correct. If we are drawing frame 5 and couldn't find a keyframe at position 5, it should find a previous keyframe and draw it (for example, keyframe 3), not just skip it.

I will send a pull request.

Reply:
Ah that's true 👍